src/rpc/RPCHelpers.cpp

```diff
@@ -1383,7 +1383,7 @@ parseBook(boost::json::object const& request)
     ripple::Currency get_currency;
     if (!ripple::to_currency(
             get_currency, taker_gets["currency"].as_string().c_str()))
-        return Status{Error::rpcINVALID_PARAMS, "badTakerGetsCurrency"};
+        return Status{Error::rpcDST_AMT_MALFORMED, "badTakerGetsCurrency"};
```

Comment:
There is a block above that does a very similar thing for `takerPaysCurrency`. Can you please verify with @ManiMounikaKunasani whether it also needs to change or should stay as `rpcINVALID_PARAMS`. I feel like it would have to change the same way.

Reply:
We definitely need to make this same change for `takerPaysCurrency`